AK/InsertionSort.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2022, Marc Luqué <marc.luque@outlook.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Concepts.h>
+#include <AK/StdLibExtras.h>
+
+namespace AK {
+
+// Standard Insertion Sort, with `end` inclusive!
+template<typename Collection, typename Comparator, typename T = decltype(declval<Collection>()[declval<int>()])>
+void insertion_sort(Collection& col, int start, int end, Comparator comparator) requires(Indexable<Collection, T>)
```

Comment:
Why int?

Reply:
ah right, I was wondering as well why Marc did that but didn't remember to change it.